Implementing __setattr__ errors

## 🐛 Bug Reports

When reporting a bug, please provide the following information. If this is not a bug report you can just discard this template.

### 🌍 Environment

 - Your operating system and version: NixOS 19.09
 - Your python version: 3.7
 - How did you install python (e.g. apt or pyenv)? Did you use a virtualenv?: nix-shell
 - Your rust version (`rustc --version`): rustc 1.38.0-nightly (311376d30 2019-07-18)
 - Are you using the latest pyo3 version? No

### 💥 Reproducing

```
#[pyclass(name=Test)]
pub struct PyTest {
    foo: usize,
}

#[pyproto]
impl PyObjectProtocol for PyTest {
    fn __setattr__(&mut self, name: String, value: String) -> PyResult<()> {
        Ok(())
    }
}
```

Fails with:

```
error[E0437]: type `Success` is not a member of trait `pyo3::class::basic::PyObjectSetAttrProtocol`
   --> src/sentence.rs:251:1
    |
251 | #[pyproto]
    | ^^^^^^^^^^ not a member of trait `pyo3::class::basic::PyObjectSetAttrProtocol`

error: aborting due to previous error
```

`__getattr__` works fine on the other hand.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 * Fix handling of invalid utf-8 sequences in `PyString::as_bytes` [#639](https://github.com/PyO3/pyo3/pull/639)
 and `PyString::to_string_lossy` [#642](https://github.com/PyO3/pyo3/pull/642).
+* Fix proc-macro definition of PySetAttrProtocol. [#645](https://github.com/PyO3/pyo3/pull/645)
 
 ## [0.8.1]
 

--- a/pyo3-derive-backend/src/defs.rs
+++ b/pyo3-derive-backend/src/defs.rs
@@ -25,7 +25,7 @@ pub const OBJECT: Proto = Proto {
             name: "__setattr__",
             arg1: "Name",
             arg2: "Value",
-            pyres: true,
+            pyres: false,
             proto: "pyo3::class::basic::PyObjectSetAttrProtocol",
         },
         MethodProto::Binary {

